src/components/forms/ActionForm.jsx

```diff
@@ -71,11 +73,18 @@ export default class ActionForm extends React.Component {
                     initialValue={ action.title } maxLength={ 300 }/>
                 <TextArea labelMsg="forms.action.info" name="info_text"
                     initialValue={ action.info_text }/>
-
+                <TextInput labelMsg="forms.action.url" name="url"
+                    initialValue={ action.url } maxLength={ 500 }
+                    onBlur={ this.onChangeUrl.bind(this) }/>
             </Form>
         );
     }
 
+    onChangeUrl(ev) {
+        const url = ev.target.value;
+        ev.target.value = normalizeUrl(url);
```

Comment:
This does not feel very "reacty". You should never change an input field like this, especially if it's a "controlled" input, i.e. an input that get it's value from the declarative JSX, as is the case in TextInput.jsx:28.